axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java

```diff
@@ -312,35 +309,79 @@ public void getFirstToken(GetFirstTokenRequest request, StreamObserver<TrackingT
 
     public void getLastToken(GetLastTokenRequest request, StreamObserver<TrackingToken> responseObserver0) {
         ForwardingStreamObserver<TrackingToken> responseObserver = new ForwardingStreamObserver<>(logger, "getLastToken", responseObserver0);
-        checkConnection(contextProvider.getContext(), responseObserver).ifPresent(client ->
+        checkConnection(contextProvider.getContext(), responseObserver, false).ifPresent(client ->
                                                             client.getLastToken(contextProvider.getContext(), request, responseObserver)
         );
     }
 
     public void getTokenAt(GetTokenAtRequest request, StreamObserver<TrackingToken> responseObserver0) {
         ForwardingStreamObserver<TrackingToken> responseObserver = new ForwardingStreamObserver<>(logger, "getTokenAt", responseObserver0);
-        checkConnection(contextProvider.getContext(), responseObserver)
+        checkConnection(contextProvider.getContext(), responseObserver, false)
                 .ifPresent(client ->client.getTokenAt(contextProvider.getContext(), request, responseObserver)
         );
     }
 
     public void readHighestSequenceNr(ReadHighestSequenceNrRequest request,
                                       StreamObserver<ReadHighestSequenceNrResponse> responseObserver0) {
         ForwardingStreamObserver<ReadHighestSequenceNrResponse> responseObserver = new ForwardingStreamObserver<>(logger, "readHighestSequenceNr", responseObserver0);
-        checkConnection(contextProvider.getContext(), responseObserver)
+        checkConnection(contextProvider.getContext(), responseObserver, false)
                 .ifPresent(client -> client.readHighestSequenceNr(contextProvider.getContext(), request, responseObserver)
         );
     }
 
-
     public StreamObserver<QueryEventsRequest> queryEvents(StreamObserver<QueryEventsResponse> responseObserver0) {
-        ForwardingStreamObserver<QueryEventsResponse> responseObserver = new ForwardingStreamObserver<>(logger, "queryEvents", responseObserver0);
-        return checkConnection(contextProvider.getContext(), responseObserver).map(client -> client.queryEvents(contextProvider.getContext(), responseObserver)).orElse(null);
+        String context = contextProvider.getContext();
+        EventStore leaderEventStore = eventStoreLocator.getEventStore(context, false);
+        EventStore localEventStore = eventStoreLocator.getEventStore(context, true);
```

Comment:
The name of this variable could be misleading, as the method may not return the local store but the remote one, according to the documentation.

Reply:
You are right! But I'm not sure about the name :) `maybeLocalEventStore`? 😄